lstchain/datachecks/dl1_checker.py

```diff
@@ -0,0 +1,1081 @@
+"""
+Functions to check the contents ofLST DL1 files and associated muon ring files
+"""
+
+__all__ = [
+    'check_dl1',
+    'process_dl1_file',
+    'plot_datacheck',
+    'plot_trigger_types',
+    'plot_mean_and_stddev',
+    'merge_dl1datacheck_files'
+]
+
+import h5py
+import logging
+import matplotlib.colors as colors
+import matplotlib.dates as dates
+import matplotlib.pyplot as plt
+import matplotlib.ticker as mtick
+import numpy as np
+import os
+import pandas as pd
+import tables
+
+from astropy import units as u
+from astropy.table import Table, vstack
+from ctapipe.coordinates import EngineeringCameraFrame
+from ctapipe.instrument import CameraGeometry
+from ctapipe.io import HDF5TableWriter
+from ctapipe.visualization import CameraDisplay
+from datetime import datetime
+from lstchain.datachecks.containers import DL1DataCheckContainer
+from lstchain.datachecks.containers import DL1DataCheckHistogramBins
+from lstchain.io.io import dl1_params_lstcam_key
+from lstchain.paths import parse_datacheck_dl1_filename, parse_dl1_filename, \
+    run_to_muon_filename, run_to_datacheck_dl1_filename
+from matplotlib.backends.backend_pdf import PdfPages
+from multiprocessing import Pool
+from pathlib import Path
+from scipy.stats import poisson, sem
+
+
+def check_dl1(filenames, output_path, max_cores=4, create_pdf=False):
+    """
+
+    Parameters
+    ----------
+    filenames: string, Path, or a list of them, _sorted_ (by growing subrun
+    index). Name(s) of the input DL1 .h5 file(s)
+    output_path: directory where output will be written
+    max_cores: maximum number of processes that the function will spawn (each
+    processing a different subrun)
+
+    Returns
+    -------
+    None
+
+    """
+
+    logger = logging.getLogger(__name__)
+
+    # convert to list if it is not yet a a list:
+    if not isinstance(filenames, list):
+        filenames = [filenames]
+
+    # Obtain the names of the corresponding muon .fits files, assumed to be
+    # in the same directory as the DL1 event files:
+    muon_filenames = []
+    for filename in filenames:
+        dl1 = parse_dl1_filename(filename)
+        muon_filenames.append(run_to_muon_filename(dl1.tel_id, dl1.run,
```

Comment:
Probably the input directory here is missing? so `filename.parent / run_to_muon_filename` ? if `filename` is a `Path` ( then it should probably be called path or `infile_path` or similar.

Reply:
> Probably the input directory here is missing? so `filename.parent / run_to_muon_filename` ? if `filename` is a `Path` ( then it should probably be called path or `infile` or similar.

Yep, will sove it now... I should not test always on the working directory :(

Reply:
> > Probably the input directory here is missing? so `filename.parent / run_to_muon_filename` ? if `filename` is a `Path` ( then it should probably be called path or `infile` or similar.
> 
> Yep, will sove it now... I should not test always on the working directory :(

Ok, done. "This should fix it, episode 134". Daniel please try it. You can perhaps test just the call to check_dl1 like it is done in r0_to_dl1